esi/src/config.rs

```diff
@@ -5,16 +5,20 @@
 /// let config = esi::Configuration::default()
 ///     .with_namespace("app");
 /// ```
+#[allow(clippy::return_self_not_must_use)]
```

Comment:
I'm not sure this is necessary. If I remove it I don't get any warnings. This appears to only be needed if `self` is taken as a function argument and not used, but in all cases here it is used.
```suggestion
```

Reply:
It's not in a Default `clippy` group, that's why you get no warning.
https://rust-lang.github.io/rust-clippy/master/index.html#/return_self_not_must_use

it's harmless otherwise